dandi/metadata.py

```diff
@@ -59,64 +68,89 @@ def get_metadata(path: Union[str, Path]) -> Optional[dict]:
     -------
     dict
     """
+
+    from .files import bids, dandi_file, find_bids_dataset_description
+
     # when we run in parallel, these annoying warnings appear
     ignore_benign_pynwb_warnings()
-    path = str(path)  # for Path
+    path = os.path.abspath(str(path))  # for Path
     meta = dict()
 
-    if op.isdir(path):
+    if op.isdir(path) and not path.endswith(tuple(ZARR_EXTENSIONS)):
         try:
             dandiset = Dandiset(path)
             return cast(dict, dandiset.metadata)
         except ValueError as exc:
             lgr.debug("Failed to get metadata for %s: %s", path, exc)
             return None
 
-    if nwb_has_external_links(path):
-        raise NotImplementedError(
-            f"NWB files with external links are not supported: {path}"
-        )
-
-    # First read out possibly available versions of specifications for NWB(:N)
-    meta["nwb_version"] = get_nwb_version(path)
-
-    # PyNWB might fail to load because of missing extensions.
-    # There is a new initiative of establishing registry of such extensions.
-    # Not yet sure if PyNWB is going to provide "native" support for needed
-    # functionality: https://github.com/NeurodataWithoutBorders/pynwb/issues/1143
-    # So meanwhile, hard-coded workaround for data types we care about
-    ndtypes_registry = {
-        "AIBS_ecephys": "allensdk.brain_observatory.ecephys.nwb",
-        "ndx-labmetadata-abf": "ndx_dandi_icephys",
-    }
-    tried_imports = set()
-    while True:
-        try:
-            meta.update(_get_pynwb_metadata(path))
-            break
-        except KeyError as exc:  # ATM there is
-            lgr.debug("Failed to read %s: %s", path, exc)
-            res = re.match(r"^['\"\\]+(\S+). not a namespace", str(exc))
-            if not res:
-                raise
-            ndtype = res.groups()[0]
-            if ndtype not in ndtypes_registry:
-                raise ValueError(
-                    "We do not know which extension provides %s. "
-                    "Original exception was: %s. " % (ndtype, exc)
-                )
-            import_mod = ndtypes_registry[ndtype]
-            lgr.debug("Importing %r which should provide %r", import_mod, ndtype)
-            if import_mod in tried_imports:
-                raise RuntimeError(
-                    "We already tried importing %s to provide %s, but it seems it didn't help"
-                    % (import_mod, ndtype)
-                )
-            tried_imports.add(import_mod)
-            __import__(import_mod)
-
-    meta["nd_types"] = get_neurodata_types(path)
+    # We assume that non-NWB data is BIDS.
```

Comment:
why?  moreover .nwb files could be part of a BIDS dandiset and then we just need to validate both according to it being NWB (our validator using nwbinspector) **and** being part of BIDS (bids validation).  So decision making shouldn't be solely based on the extension for `else:`.  Pretty much you can validate here with such `if` but then next codeblock should 

- determine if dandiset in BIDS layout (has dataset_description.json on top), and if yes -- validate against BIDS (with possibly BIDS subdatasets under `rawdata/` etc  - so search for such files below), if not -- assume DANDI layout (to have validation actually implemented -- #1163, attn @jwodder).

Reply:
and we can have files which aren't .nwb and not BIDS, could be video files associated with .nwb and still be just DANDI layout.... so as to me -- such assumption is way too incorrect.  Do you think you could refactor the way I suggested?

Reply:
@yarikoptic good now?

Reply:
it changed nothing if I see it right... I have suggested code change in https://github.com/dandi/dandi-cli/pull/1164/files#r1046140228 . I will resolve this conversation -- let's continue there